python/ray/data/tests/test_groupby_e2e.py

```diff
@@ -1209,7 +1209,8 @@ def func(x, y):
         {"x_add_5": 25},
         {"x_add_5": 25},
     ]
-    assert "MapBatches(func)" in ds.__repr__()
+    plan_repr = ds._plan.get_plan_as_string(type(ds))
+    assert "MapBatches(func)" in plan_repr
 
```

Comment:
With this and `test_map` -- would it make sense to assert against `Dataset.explain()` instead?